.github/workflows/test-and-publish-nightly.yml

```diff
@@ -51,64 +51,100 @@ jobs:
             java-version: 11
             java-distribution: corretto
             container: "public.ecr.aws/async-profiler/asprof-builder-arm:latest"
+          - runson:
+              display: amazonlinux-2023 # Use dash instead of colon, because GHA artifacts cannot contain colons
+              name: ubuntu-latest
+            java-version: 11
+            java-distribution: corretto
+            container: amazonlinux:2023
+          - runson:
+              display: amazonlinux-2
+              name: ubuntu-latest
+            java-version: 11
+            java-distribution: corretto
+            # Make sure this is in sync with java-version/distribution for UX purposes
+            java-package: java-11-amazon-corretto
+            container: amazonlinux:2
     runs-on: ${{ matrix.runson.name }}
     container:
       image: ${{ matrix.container }}
       options: --privileged
     name: "test (${{ matrix.runson.display }}, ${{ matrix.java-distribution }} ${{ matrix.java-version }})"
     steps:
+      - name: Setup
+        run: |
+          if [[ "${{ matrix.container }}" == "amazonlinux"* ]]; then
+              yum update -y
+              yum install -y make gcc-c++ libstdc++-devel libstdc++-static util-linux tar git patchelf
+          fi
       - name: Setup Java
         uses: actions/setup-java@v4
+        if: matrix.runson.display != 'amazonlinux-2'
         with:
           distribution: ${{ matrix.java-distribution }}
           java-version: ${{ matrix.java-version }}
           architecture: ${{ matrix.architecture }}
+      - name: Setup Java on amazonlinux-2
+        if: matrix.runson.display == 'amazonlinux-2'
+        run: yum install -y ${{ matrix.java-package }}
       - name: Checkout sources
+        if: matrix.runson.display != 'amazonlinux-2'
         uses: actions/checkout@v4
+      - name: Checkout sources on amazonlinux-2
+        if: matrix.runson.display == 'amazonlinux-2'
+        run: |
+          git clone https://${{ github.token }}@github.com/${{ github.repository }}.git .
```

Comment:
We can use `--depth=1` for faster cloning, as we don't need the repo history.

Reply:
Actually the reason why I've done it this way is that I foreseen a subsequent `git checkout ${{ github.sha }}` which I forgot to put in the diff. We cannot use `--branch` because that could result in some nasty race conditions

Reply:
Let's use this to avoid checking out the entire history:

```
git init
git remote add origin https://github.com/async-profiler/async-profiler
git fetch --depth 1 origin b9f3456f892088e9aac8cadbb3a9f4a86a7f3654
git checkout b9f3456f892088e9aac8cadbb3a9f4a86a7f3654
```

Reply:
Thanks!